src/Datadog.Trace/Agent/TraceRequestDecorator.cs

```diff
@@ -0,0 +1,13 @@
+namespace Datadog.Trace.Agent
+{
+    internal static class TraceRequestDecorator
```

Comment:
This isn't a big deal, but without a full Decorator pattern, this feels a little weird to me. Maybe this could be `IApiRequestExtensions`?